### PR TITLE
feat(share/proof)!: use nmt definition of Proof

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
remove the node's definition of pb proof in favour of the nmt one.
Resolves https://github.com/celestiaorg/celestia-node/issues/2512
## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by Wondertan — COMMENTED** [review]
I wonder if it's really breaking if the structure of Proof didn't change. Proto is usually smart about such things. Like changing var name is not breaking.

**Comment by Wondertan** [comment]
cc @vgonkivs

Just so we don't break things for no reason.

**Comment by vgonkivs** [comment]
During our internal discussion, I understood you incorrectly answering the structure was not changed(I meant nmt), I forgot to tell you that the pb struct has 1 extra field - IsMaxNamespaceIgnored. 

PS. found one more place in the code that still uses our internal proof. Will remove in the next commit

**Review by Wondertan — APPROVED** [review]

File: share/p2p/shrexnd/params.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/celestiaorg/celestia-node/share/p2p"
 )
 
-const protocolString = "/shrex/nd/v0.0.2"
```
Comment (renaynay) [inline share/p2p/shrexnd/params.go:11]: It would be nice to merge this PR plus the #2444 together so that we can break shrexnd one time.

**Review by renaynay — APPROVED** [review]
Approving but see comment please - cc @Wondertan @walldiss

**Review by Wondertan — DISMISSED** [review]

**Comment by Wondertan** [comment]
@vgonkivs, linter

**Comment by vgonkivs** [comment]
There is smth that is not related to my PR. Will restart the job

**Review by Wondertan — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**